tests/test_tcp_models.py

```diff
@@ -297,6 +297,44 @@ async def test_inverter_tcp() -> None:
     await server_exit
 
 
+async def test_inverter_tcp_no_firmware() -> None:
+    """Test request from an Inverter without firmware reply - TCP source."""
+    serial_number = 987654321
+
+    (server_exit, port) = tcp_server(serial_number, "tcp_reply_no_firmware.data")
+
+    client = OmnikInverter(
+        host="localhost",
+        source_type="tcp",
+        serial_number=serial_number,
+        tcp_port=port,
+    )
+
+    inverter: Inverter = await client.inverter()
+
+    assert inverter
+    assert inverter.solar_rated_power is None
+    assert inverter.solar_current_power == 0
+
+    assert inverter.model is None
+    assert inverter.serial_number == "NLDN202013212035"
+    assert inverter.temperature == 23.400000000000002
+    assert inverter.dc_input_voltage == [118.30000000000001, 0.0, None]
+    assert inverter.dc_input_current == [0.0, 15.100000000000001, None]
```

Comment:
Perhaps a data / parser error? Seems odd to have an unconnected string report a positive Amperage, but no Voltage.

Reply:
@PeeVv I'm using your dump from https://github.com/robbinjanssen/home-assistant-omnik-inverter/issues/142#issuecomment-1507430793 for this test-case if that's okay. If not I recommend you to remove the public link and I'll anonymize this packet :)

Reply:
> @PeeVv I'm using your dump from [robbinjanssen/home-assistant-omnik-inverter#142 (comment)](https://github.com/robbinjanssen/home-assistant-omnik-inverter/issues/142#issuecomment-1507430793) for this test-case if that's okay. If not I recommend you to remove the public link and I'll anonymize this packet :)

Fine by me! You can use it for whatever you want.